app/src/lib/git/pull.ts

```diff
@@ -110,3 +120,23 @@ export async function pull(
     throw new GitError(result, args)
   }
 }
+
+/**
+ * Defaults the pull default for divergent paths to try to fast forward and if
+ * not perform a merge. Aka uses the flag --ff
+ *
+ * It checks whether the user has a config set for this already, if so, no need for
+ * default.
+ */
+async function getDefaultPullDivergentBranchArguments(
+  repository: Repository
+): Promise<ReadonlyArray<string>> {
+  const pullRebase = await getConfigValue(repository, 'pull.rebase')
+  const pullFF = await getConfigValue(repository, 'pull.ff')
```

Comment:
Given that this is in such a critical path of the app (pulling) and given that we're doing a major release tomorrow I think it's prudent to be a bit conservative here and guard against this lookup failing. If it does we can assume it's set (which will cause the error described in the PR which isn't great but it's better than failing completely)

```suggestion
  const pullFF = await getConfigValue(repository, 'pull.ff').catch(e => {
    log.error("Couldn't read 'pull.ff' config", e)
    return null
  })
```

Reply:
Not sure if you have a strong preference to the `.catch` vs in a `try/catch`. But, the logic we need would require return a faux response such as "true" (really just not null)... just a personal preference of mine not to introduce fake responses and just acknowledge the failure in the try/catch. But, not terribly strong willed on this if you really refer the `.catch` syntax (or there is a functionally compelling reason to prefer it).

Reply:
I don't have a strong preference no. In this case I like having catch return a value that the subsequent logic can then handle as if it was a default value. That way there's only one source of truth for what the "no value" case should return. But this is very much on a case-by-case basis and could just as easily have been accomplished with a variable.

All that said this is likely how I would have written that method had the stylistic choice been entirely up to me 😄 

```ts
const getDefaultPullDivergentBranchArguments = (repository: Repository) =>
  getConfigValue(repository, 'pull.ff')
    .catch(e => {
      log.error("Couldn't read 'pull.ff' config", e)
      return null
    })
    .then(ff => (ff === null ? ['ff'] : []))
```